src/dvc_data/hashfile/build.py

```diff
@@ -150,6 +150,11 @@ def _build_tree(
                 tree_meta.size += meta.size or 0
                 tree_meta.nfiles += 1
 
+    if not tree_meta.nfiles:
+        # This will raise FileNotFoundError if it is a
+        # broken symlink or TreeError
+        next(iter(fs.ls(path)))
+
```

Comment:
Proper solution is to fix fsspec's walk and find to not swallow FIleNotFoundError but actually raise if they detect a broken symlink kind of case.